addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveProtectionConverter.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zwave.internal.converter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.zwave.handler.ZWaveControllerHandler;
+import org.openhab.binding.zwave.handler.ZWaveThingChannel;
+import org.openhab.binding.zwave.internal.protocol.SerialMessage;
+import org.openhab.binding.zwave.internal.protocol.ZWaveNode;
+import org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveCommandClass;
+import org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveProtectionCommandClass;
+import org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveProtectionCommandClass.LocalProtectionType;
+import org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveProtectionCommandClass.RfProtectionType;
+import org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveProtectionCommandClass.Type;
+import org.openhab.binding.zwave.internal.protocol.event.ZWaveCommandClassValueEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ZWaveProtectionConverter class. Converters between binding items and the Z-Wave API for protection.
+ *
+ * @author Jorg de Jong
+ */
+public class ZWaveProtectionConverter extends ZWaveCommandClassConverter {
+
+    private static final Logger logger = LoggerFactory.getLogger(ZWaveProtectionConverter.class);
+
+    /**
+     * Constructor. Creates a new instance of the {@link ZWaveConverterBase} class.
+     *
+     */
+    public ZWaveProtectionConverter(ZWaveControllerHandler controller) {
+        super(controller);
+    }
+
+    @Override
+    public State handleEvent(ZWaveThingChannel channel, ZWaveCommandClassValueEvent event) {
+        String type = channel.getArguments().get("type");
+
+        // Don't trigger event if this item is bound to another type
+        if (type != null && !event.getType().equals(Type.valueOf(type))) {
+            return null;
+        }
+
+        Enum<?> e = (Enum<?>) event.getValue();
+        return new DecimalType(e.ordinal());
+    }
+
+    @Override
+    public List<SerialMessage> receiveCommand(ZWaveThingChannel channel, ZWaveNode node, Command command) {
+        String type = channel.getArguments().get("type");
+
+        ZWaveProtectionCommandClass commandClass = (ZWaveProtectionCommandClass) node
+                .resolveCommandClass(ZWaveCommandClass.CommandClass.PROTECTION, channel.getEndpoint());
```

Comment:
I think resolveCommandClass can return null if the class isn't found so we probably should check for this with -:
```
if(commandClass == null) {
    return null;
}
```